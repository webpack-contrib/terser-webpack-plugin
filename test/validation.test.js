import TerserPlugin from '../src';

it('validation', () => {
  /* eslint-disable no-new */
  expect(() => {
    new TerserPlugin({ test: /foo/ });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ test: [/foo/] });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ include: /foo/ });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ include: [/foo/] });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ exclude: /foo/ });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ exclude: [/foo/] });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ doesntExist: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new TerserPlugin({ cache: true });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ cache: false });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ cache: 'path/to/cache/directory' });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ cache: {} });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new TerserPlugin({ cacheKeys() {} });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ parallel: true });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ parallel: false });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ parallel: 2 });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ parallel: '2' });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new TerserPlugin({ parallel: {} });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new TerserPlugin({ sourceMap: true });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ sourceMap: false });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ sourceMap: 'true' });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new TerserPlugin({ minify() {} });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ terserOptions: null });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new TerserPlugin({ terserOptions: {} });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({
      terserOptions: {
        // eslint-disable-next-line no-undefined
        ecma: undefined,
        warnings: false,
        parse: {},
        compress: {},
        mangle: true,
        module: false,
        output: null,
        toplevel: false,
        nameCache: null,
        ie8: false,
        keep_classnames: false,
        keep_fnames: false,
        safari10: false,
      },
    });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ terserOptions: { emca: 5 } });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ extractComments: true });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ extractComments: false });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ extractComments: /comment/ });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ extractComments() {} });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ warningsFilter() {} });
  }).not.toThrow();

  expect(() => {
    new TerserPlugin({ warningsFilter: true });
  }).toThrowErrorMatchingSnapshot();
  /* eslint-enable no-new */
});
