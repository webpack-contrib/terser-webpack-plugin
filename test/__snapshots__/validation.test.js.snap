// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"Terser Plugin Invalid Options

options.test should pass \\"instanceof\\" keyword validation
options.test should be string
options.test should match some schema in anyOf
options.test should be array
options.test should match some schema in anyOf
"
`;

exports[`validation 2`] = `
"Terser Plugin Invalid Options

options.test should pass \\"instanceof\\" keyword validation
options.test should be string
options.test should match some schema in anyOf
options.test.0 should pass \\"instanceof\\" keyword validation
options.test.0 should be string
options.test.0 should match some schema in anyOf
options.test.0 should match some schema in anyOf
options.test should match some schema in anyOf
"
`;

exports[`validation 3`] = `
"Terser Plugin Invalid Options

options.include should pass \\"instanceof\\" keyword validation
options.include should be string
options.include should match some schema in anyOf
options.include should be array
options.include should match some schema in anyOf
"
`;

exports[`validation 4`] = `
"Terser Plugin Invalid Options

options.include should pass \\"instanceof\\" keyword validation
options.include should be string
options.include should match some schema in anyOf
options.include.0 should pass \\"instanceof\\" keyword validation
options.include.0 should be string
options.include.0 should match some schema in anyOf
options.include.0 should match some schema in anyOf
options.include should match some schema in anyOf
"
`;

exports[`validation 5`] = `
"Terser Plugin Invalid Options

options.exclude should pass \\"instanceof\\" keyword validation
options.exclude should be string
options.exclude should match some schema in anyOf
options.exclude should be array
options.exclude should match some schema in anyOf
"
`;

exports[`validation 6`] = `
"Terser Plugin Invalid Options

options.exclude should pass \\"instanceof\\" keyword validation
options.exclude should be string
options.exclude should match some schema in anyOf
options.exclude.0 should pass \\"instanceof\\" keyword validation
options.exclude.0 should be string
options.exclude.0 should match some schema in anyOf
options.exclude.0 should match some schema in anyOf
options.exclude should match some schema in anyOf
"
`;

exports[`validation 7`] = `
"Terser Plugin Invalid Options

options.chunkFilter should pass \\"instanceof\\" keyword validation
"
`;

exports[`validation 8`] = `
"Terser Plugin Invalid Options

options.cache should be boolean
options.cache should be string
options.cache should match some schema in anyOf
"
`;

exports[`validation 9`] = `
"Terser Plugin Invalid Options

options.cacheKeys should pass \\"instanceof\\" keyword validation
"
`;

exports[`validation 10`] = `
"Terser Plugin Invalid Options

options.parallel should be boolean
options.parallel should be integer
options.parallel should match some schema in anyOf
"
`;

exports[`validation 11`] = `
"Terser Plugin Invalid Options

options.parallel should be boolean
options.parallel should be integer
options.parallel should match some schema in anyOf
"
`;

exports[`validation 12`] = `
"Terser Plugin Invalid Options

options.sourceMap should be boolean
"
`;

exports[`validation 13`] = `
"Terser Plugin Invalid Options

options.minify should pass \\"instanceof\\" keyword validation
"
`;

exports[`validation 14`] = `
"Terser Plugin Invalid Options

options.terserOptions should be object
"
`;

exports[`validation 15`] = `
"Terser Plugin Invalid Options

options.extractComments should be boolean
options.extractComments should be string
options.extractComments should pass \\"instanceof\\" keyword validation
options.extractComments should pass \\"instanceof\\" keyword validation
options.extractComments.condition should be boolean
options.extractComments.condition should be string
options.extractComments.condition should pass \\"instanceof\\" keyword validation
options.extractComments.condition should pass \\"instanceof\\" keyword validation
options.extractComments.condition should match some schema in anyOf
options.extractComments should match some schema in anyOf
"
`;

exports[`validation 16`] = `
"Terser Plugin Invalid Options

options.extractComments should be boolean
options.extractComments should be string
options.extractComments should pass \\"instanceof\\" keyword validation
options.extractComments should pass \\"instanceof\\" keyword validation
options.extractComments.filename should be string
options.extractComments.filename should pass \\"instanceof\\" keyword validation
options.extractComments.filename should match some schema in anyOf
options.extractComments should match some schema in anyOf
"
`;

exports[`validation 17`] = `
"Terser Plugin Invalid Options

options.extractComments should be boolean
options.extractComments should be string
options.extractComments should pass \\"instanceof\\" keyword validation
options.extractComments should pass \\"instanceof\\" keyword validation
options.extractComments.banner should be boolean
options.extractComments.banner should be string
options.extractComments.banner should pass \\"instanceof\\" keyword validation
options.extractComments.banner should match some schema in anyOf
options.extractComments should match some schema in anyOf
"
`;

exports[`validation 18`] = `
"Terser Plugin Invalid Options

options.extractComments should be boolean
options.extractComments should be string
options.extractComments should pass \\"instanceof\\" keyword validation
options.extractComments should pass \\"instanceof\\" keyword validation
options.extractComments should NOT have additional properties
options.extractComments should match some schema in anyOf
"
`;

exports[`validation 19`] = `
"Terser Plugin Invalid Options

options.warningsFilter should pass \\"instanceof\\" keyword validation
"
`;

exports[`validation 20`] = `
"Terser Plugin Invalid Options

options should NOT have additional properties
"
`;
